doc/source/data/transforming-data.rst

```diff
@@ -144,6 +175,34 @@ To configure the batch type, specify ``batch_format`` in
                 .map_batches(drop_nas, batch_format="pandas")
             )
 
+The user defined function passed to :meth:`~ray.data.Dataset.map_batches` is more flexible. As batches
+can be represented in multiple ways (more on this in :ref:`Configuring batch format <configure_batch_format>`), so the function should be of type
+``Callable[DataBatch, DataBatch]``, where ``DataBatch = Union[pd.DataFrame, Dict[str, np.ndarray]]``. In
+other words, your function should take as input and output a batch of data which can be represented as a
+pandas DataFrame or a dictionary with string keys and NumPy ndarrays values. Your function doesn't need
+to return a batch in the same format as its input, so you could input a pandas DataFrame and output a
+dictionary of NumPy ndarrays. For example, your function might look like:
```

Comment:
I prefer not to mention this. I don't think it's useful in practice. It's only because our current implementation allows it. We may want to enforce the same format in the future.